### PR TITLE
Minor: clean up TODO comments in unnest.slt

### PR DESCRIPTION
**Pull request by goldmedal**
## Which issue does this PR close?

Closes #.

## What changes are included in this PR?
Remove some legacy TODO comments and refine the tests for unnest.

## Are these changes tested?
the original CI

## Are there any user-facing changes?
no

### CODE REVIEW

File: datafusion/sqllogictest/test_files/unnest.slt [hunk 1]
```diff
@@ -524,10 +533,19 @@ select unnest(column1) from (select * from (values([1,2,3]), ([4,5,6])) limit 1
 5
 6
 
-## FIXME: https://github.com/apache/datafusion/issues/11198
```
Comment (goldmedal) [inline datafusion/sqllogictest/test_files/unnest.slt:527]: This issue has been closed. The error assertion makes sense to me. I also added another positive test at L539

File: datafusion/sqllogictest/test_files/unnest.slt [hunk 2]
```diff
@@ -511,10 +511,19 @@ x y [30, 40, 50]
 query error DataFusion error: type_coercion\ncaused by\nThis feature is not implemented: Unnest should be rewritten to LogicalPlan::Unnest before type coercion
 select sum(unnest(generate_series(1,10)));
 
-## TODO: support unnest as a child expr
 query error DataFusion error: Internal error: unnest on struct can only be applied at the root level of select expression
 select arrow_typeof(unnest(column5)) from unnest_table;
```
Comment (goldmedal) [inline datafusion/sqllogictest/test_files/unnest.slt:515]: This error isn't a child expression issue. It's an issue about applying a function to an unnest struct expression. The error message makes sense to me. DuckDB has similar behavior:
```sql
D CREATE TABLE unnest_table
  AS VALUES
      ([1,2,3], [7], 1, [13, 14], {'c1': 1, 'c2': 2}),
      ([4,5], [8,9,10], 2, [15, 16], {'c1':3,'c2':4}),
      ([6], [11,12], 3, null, null),
      ([12], [null, 42, null], null, null, {'c1':7,'c2':8}),
      -- null array to verify the `preserve_nulls` option
      (null, null, 4, [17, 18], null)
  ;
D select typeof(unnest(col4)) from unnest_table;
Binder Error: UNNEST() on a struct column can only be applied as the root element of a SELECT expression
LINE 1: select typeof(unnest(col4)) from unnest_table
```

File: datafusion/sqllogictest/test_files/unnest.slt [hunk 3]
```diff
@@ -511,10 +511,19 @@ x y [30, 40, 50]
...
 
+query T
+select arrow_typeof(unnest(column1)) from unnest_table;
+----
+Int64
```
Comment (goldmedal) [inline datafusion/sqllogictest/test_files/unnest.slt:520]: Unnest as a child expression works fine if the input isn't a struct.

File: datafusion/sqllogictest/test_files/unnest.slt [hunk 4]
```diff
@@ -777,10 +795,13 @@ select unnest(unnest(column2)) c2, count(column3) from recursive_unnest_table gr
 [, 6] 1
 NULL 1
 
-### TODO: group by unnest struct
 query error DataFusion error: Error during planning: Projection references non\-aggregate values
 select unnest(column1) c1 from nested_unnest_table group by c1.c0;
```
Comment (goldmedal) [inline datafusion/sqllogictest/test_files/unnest.slt:799]: Although the error message is weird, asserting that this query fails makes sense to me.
DuckDB doesn't allow this SQL, too. I can't find other databases with this behavior. 🤔 

```sql
D CREATE TABLE unnest_table
  AS VALUES
      ([1,2,3], [7], 1, [13, 14], {'c1': 1, 'c2': 2}),
      ([4,5], [8,9,10], 2, [15, 16], {'c1':3,'c2':4}),
      ([6], [11,12], 3, null, null),
      ([12], [null, 42, null], null, null, {'c1':7,'c2':8}),
      -- null array to verify the `preserve_nulls` option
      (null, null, 4, [17, 18], null)
  ;
D select unnest(col4) u1 from unnest_table group by u1.c1;
Binder Error: Referenced table "u1" not found!
Candidate tables: "unnest_table"
LINE 1: ...st(col4) u1 from unnest_table group by u1.c1;
```

File: datafusion/sqllogictest/test_files/unnest.slt [hunk 5]
```diff
@@ -777,10 +795,13 @@ select unnest(unnest(column2)) c2, count(column3) from recursive_unnest_table gr
...
 
+# TODO: this query should work. see issue: https://github.com/apache/datafusion/issues/12794
+query error DataFusion error: Internal error: unnest on struct can only be applied at the root level of select expression
+select unnest(column1) c1 from nested_unnest_table
```
Comment (goldmedal) [inline datafusion/sqllogictest/test_files/unnest.slt:803]: Filed https://github.com/apache/datafusion/issues/12794 for it.

**Review by alamb — APPROVED** [review]
Thank you @goldmedal

**Comment by alamb** [comment]
🚀

**Pull request merged**